app/global/MainModule.java

```diff
@@ -84,6 +86,21 @@ protected void configure() {
     }
 
     @Singleton
+    @Provides
+    @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
+    private HealthProvider getHealthProvider(final Environment environment, final Configuration config) {
+        try {
+            final Class<? extends HealthProvider> clazz = ConfigurationHelper.getType(
+                    environment,
+                    config,
+                    "http.healthProvider");
+            return clazz.newInstance();
```

Comment:
Actually considered this and figured it could be added later. I'm actually not looking to replace the health check logic, but rather the endpoint from which it is served and what format it takes.

Anyway, your example doesn't seem to fit this case. I've put something together along the same lines as other providers using an injected injector. This delegates back to Guice the construction of the configured instance allowing its constructor to be marked as @Inject.

I'll post it shortly.